narwhal/consensus/src/bullshark.rs

```diff
@@ -35,42 +53,30 @@ impl ConsensusProtocol for Bullshark {
         let round = certificate.round();
         let mut consensus_index = consensus_index;
 
-        // We must have stored already the parents of this certiciate!
-        if round > 0 {
-            let parents = certificate.header.parents.clone();
-            if let Some(round_table) = state.dag.get(&(round - 1)) {
-                let store_parents: BTreeSet<&CertificateDigest> =
-                    round_table.iter().map(|(_, (digest, _))| digest).collect();
-
-                for parent_digest in parents {
-                    if !store_parents.contains(&parent_digest) {
-                        if round - 1 + self.gc_depth > state.last_committed_round {
-                            error!(
-                                "The store does not contain the parent of {:?}: Missing item digest={:?}",
-                                certificate, parent_digest
-                            );
-                        } else {
-                            debug!(
-                                "The store does not contain the parent of {:?}: Missing item digest={:?} (but below GC round)",
-                                certificate, parent_digest
-                            );
-                        }
-                    }
-                }
-            } else {
-                error!(
-                    "Round not present in Dag store: {:?} when looking for parents of {:?}",
-                    round - 1,
-                    certificate
-                );
-            }
-        }
+        // We must have stored already the parents of this certificate!
+        self.check_parents_exist(&certificate, state);
 
         // Add the new certificate to the local storage.
         if state.try_insert(certificate).is_err() {
             return Ok(Vec::new());
         }
 
+        if round != self.max_inserted_certificate_round && round % 2 == 0 {
+            // check when the last non successful leader election was - if it is != the round-2
+            // then report an unsuccessful election
+            let last_round = &self.last_election_round;
+
+            if !last_round.leader_found {
+                // increase the leader not found metric
```

Comment:
nit - this comment and the one below kinda spammy to me, it doesn't add any context beyond what you can already tell from the variable and method names

Reply:
You are right, initially put them there when writing the code as guidance and forgot to remove them - thanks for flagging this